### PR TITLE
[Workspace]Following pr for #5949

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

- [x] Address comments on #5949 

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ruanyl — APPROVED** [review]
Thanks for the quick updates, LGTM

**Review by ashwin-pc — APPROVED** [review]
Lets wait for the CI to pass before merging, but this PR otherwise looks good

**Pull request merged**
